pkg/controller/controller_ref_manager.go

```diff
@@ -350,7 +351,14 @@ func (m *ReplicaSetControllerRefManager) AdoptReplicaSet(rs *extensions.ReplicaS
 	// Note that ValidateOwnerReferences() will reject this patch if another
 	// OwnerReference exists with controller=true.
 	addControllerPatch := fmt.Sprintf(
-		`{"metadata":{"ownerReferences":[{"apiVersion":"%s","kind":"%s","name":"%s","uid":"%s","controller":true,"blockOwnerDeletion":true}],"uid":"%s"}}`,
+		`{
+			"metadata":
+			{	
+				"ownerReferences": [{"apiVersion":"%s","kind":"%s","name":"%s","uid":"%s","controller":true,"blockOwnerDeletion":true}],
+				"uid":"%s",
+				"finalizers": ["foregroundDeletion"]
```

Comment:
What does this mean? How does it make sense to create these with a foregroundDeletion finalizer and a default deletion propagation strategy of orphan? I never expected to see a foregroundDeletion finalizer on an object without a deletionTimestamp. @jwforres something like this might explain what you were seeing